src/core/init/manifest_update_scheduler.ts

```diff
@@ -20,103 +20,159 @@ import {
-) : Observable<never> {
+export default function manifestUpdateScheduler({
+  fetchManifest,
+  initialManifest,
+  manifestUpdateUrl,
+  maximumManifestUpdateInterval,
+  minimumManifestUpdateInterval,
+  scheduleRefresh$,
+} : IManifestUpdateSchedulerArguments) : Observable<never> {
+  // The Manifest always keeps the same Manifest
+  const { manifest } = initialManifest;
+
   function handleManifestRefresh$(
     manifestInfos: { manifest: Manifest;
                      sendingTime?: number; }): Observable<never> {
-    const { manifest, sendingTime } = manifestInfos;
-    // schedule a Manifest refresh to avoid sending too much request.
-    const manualRefresh$ = scheduleRefresh$.pipe(
-      mergeMap((delay) => {
-        const timeSinceLastRefresh = sendingTime == null ?
-                                       0 :
-                                       performance.now() - sendingTime;
-        const minInterval = Math.max(minimumManifestUpdateInterval
-                                       - timeSinceLastRefresh,
-                                     0);
-        return observableTimer(Math.max(delay - timeSinceLastRefresh,
-                                        minInterval));
-      }));
-
-    const autoRefresh$ = (() => {
-      if (manifest.lifetime == null || manifest.lifetime <= 0) {
-        return EMPTY;
-      }
-      const timeSinceRequest = sendingTime == null ?
-                                 0 :
-                                 performance.now() - sendingTime;
-      const minInterval = Math.max(minimumManifestUpdateInterval
-                                     - timeSinceRequest,
-                                   0);
-      const updateTimeout = manifest.lifetime * 1000 - timeSinceRequest;
-      return observableTimer(Math.max(updateTimeout, minInterval));
-    })();
-
-    const expired$ = manifest.expired === null ? EMPTY :
-                                                 observableFrom(manifest.expired);
+    const { sendingTime } = manifestInfos;
+
+    const manualRefresh$ = scheduleRefresh$.pipe(mergeMap(({ completeRefresh }) =>
+      startManualRefreshInterval(completeRefresh,
+                                 minimumManifestUpdateInterval,
+                                 sendingTime).pipe(mapTo(completeRefresh))));
+
+    const timeSinceRequest = sendingTime == null ? 0 :
+                                                   performance.now() - sendingTime;
+    const minInterval = Math.max(minimumManifestUpdateInterval - timeSinceRequest, 0);
+
+    const autoRefresh$ = manifest.lifetime === undefined || manifest.lifetime <= 0 ?
+      EMPTY :
+      observableTimer(Math.max(manifest.lifetime * 1000 - timeSinceRequest,
+                               minInterval)).pipe(mapTo(false));
+
+    const expired$ = manifest.expired === null ?
+      EMPTY :
+      observableTimer(minimumManifestUpdateInterval)
+        .pipe(mergeMapTo(observableFrom(manifest.expired)),
+              mapTo(true));
```

Comment:
Why map to true ?

Reply:
manifest.expired is just a promise resolving with `void`.
The Observable emitted true when it should perform a full refresh.

As it was not clear enough I'm replacing the boolean by the type `{ completeRefresh: boolean }`

Reply:
Done